src/encoding/hex.rs

```diff
@@ -0,0 +1,176 @@
+//! Data independent (i.e. constant time-ish) hexadecimal encoding and decoding
+
+// Copyright (c) 2014 Steve "Sc00bz" Thomas (steve at tobtu dot com)
```

Comment:
Hey @Sc00bz, this is derived from your code. Would you mind giving me permission to dual license it under Apache 2.0 as well? (Rust people seem to love MIT/Apache-2.0 for some reason, and that's what the rest of this project is licensed under)